lib/galaxy/webapps/galaxy/api/tools.py

```diff
@@ -110,6 +110,26 @@ def reload( self, trans, id, **kwd ):
 
     @expose_api
     @web.require_admin
+    def list_requirements(self, trans, **kwds):
+        """
+        GET /api/tools/list_requirements
+        Return list of unique requirements for all tools.
+        """
+
+        return trans.app.toolbox.tools_requirements()
+
+    @expose_api
+    @web.require_admin
+    def tool_requirements_status(self, trans, id, **kwds):
+        """
+        GET /api/tools/{tool_id}/requirement_status
```

Comment:
In fact I should probably make all_requirements return the same structure ... huh, still some more work ahead :)